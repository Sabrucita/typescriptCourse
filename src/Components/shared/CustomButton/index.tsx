import React from 'react';
import {Text, StyleSheet, Pressable} from 'react-native';

interface Props {
  onPress: any;
  text: string;
  testID?: string;
}

const CustomButton: React.FC<Props> = ({onPress, text, testID}) => {
  return (
    <Pressable style={styles.button} onPress={onPress} testID={testID}>
      <Text style={styles.buttonText}>{text}</Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#015D67',
    marginHorizontal: 50,
    marginVertical: 20,
    padding: 13,
    borderRadius: 10,
  },
  buttonText: {
    color: '#FFF',
    fontWeight: '700',
    textAlign: 'center',
    fontSize: 15,
  },
});
export default CustomButton;
