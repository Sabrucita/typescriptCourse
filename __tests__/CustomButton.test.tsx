/**
 * @format
 */

import 'react-native';
import React from 'react';
import CustomButton from '../src/Components/shared/CustomButton';

// Note: test renderer must be required after react-native.
import {render, waitFor} from '@testing-library/react-native';

jest.useFakeTimers();

describe('CustomButton', () => {
  it('renders correctly', async () => {
    const {getByText, queryByTestId, toJSON} = render(
      <CustomButton
        onPress={() => {}}
        text="Test text"
        testID="example-button"
      />,
    );

    await waitFor(() => expect(queryByTestId('example-button')).toBeTruthy());

    expect(getByText('Test text')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
