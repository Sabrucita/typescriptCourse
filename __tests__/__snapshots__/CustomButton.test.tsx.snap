// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#015D67",
      "borderRadius": 10,
      "marginHorizontal": 50,
      "marginVertical": 20,
      "padding": 13,
    }
  }
  testID="example-button"
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 15,
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
  >
    Test text
  </Text>
</View>
`;
